src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,186 @@
+                throw new Error(pythonVersionIncompatiable);
+            }
+            throw new Error(localize('pythonVersionRequired', 'Python 3.6 is required to create a Python Function project.'));
+        }
+
+    }
+
+    private async ensureVirtualEnviornment(): Promise<void> {
+        const funcEnv: string = 'func_env';
+        let pythonVenvPath: string;
+        const createButton: MessageItem = { title: 'Create' };
+        const existingButton: MessageItem = { title: 'Existing' };
+        const venvRequired: string = localize('venvRequired', 'You must be running in a Python virtual environment to create a Python Function project. Create a new one or use an existing?');
+        const input: MessageItem = await ext.ui.showWarningMessage(venvRequired, { modal: true }, createButton, existingButton);
+        if (input === createButton) {
+            // if there is no func_env, create one as it's required for Python functions
+            await cpUtils.executeCommand(ext.outputChannel, this.functionAppPath, this.pythonAlias, '-m', 'venv', 'func_env');
+            pythonVenvPath = funcEnv;
+        } else {
+            const options: OpenDialogOptions = {
+                canSelectFiles: false,
+                canSelectFolders: true,
+                canSelectMany: false,
+                defaultUri: workspace.workspaceFolders && workspace.workspaceFolders.length > 0 ? workspace.workspaceFolders[0].uri : undefined,
+                openLabel: localize('select', 'Select')
+            };
+            const venvFspath: string = (await ext.ui.showOpenDialog(options))[0].fsPath;
+            pythonVenvPath = path.relative(this.functionAppPath, venvFspath);
+        }
+        await updateWorkspaceSetting(this.pythonVenvPathSetting, pythonVenvPath, this.functionAppPath);
+    }
+
+    private async getVenvActivatePath(platform: Platform): Promise<string> {
+        const venvPath: string | undefined = getFuncExtensionSetting(this.pythonVenvPathSetting, this.functionAppPath);
+        if (venvPath) {
+            switch (platform) {
+                case Platform.Windows:
+                    return path.join('.', venvPath, 'Scripts', 'activate');
+                case Platform.MacOS:
+                default:
+                    // assuming OSX and Linux use the same path -- need to some testing
+                    return path.join('.', venvPath, 'bin', 'activate');
+            }
+
+        } else {
+            // if there is no venv path, prompt user and recursively call function again
+            await this.ensureVirtualEnviornment();
+            return await this.getVenvActivatePath(platform);
+        }
+    }
+
+    private async createPythonProject(): Promise<void> {
+        switch (process.platform) {
+            case Platform.Windows:
+                await cpUtils.executeCommand(ext.outputChannel, this.functionAppPath, `${await this.getVenvActivatePath(Platform.Windows)} && func init ./ --worker-runtime python`);
```

Comment:
Shouldn't this be `|` instead of `&&`?

Reply:
No, I think cpUtils spawns a bash terminal so `|` doesn't work even in Windows.